os/drivers/usbhost/uvc/usbhost_uvc_video.c

```diff
@@ -0,0 +1,1190 @@
+			break;
+		}
+
+		bandwidth = probe->dwMaxPayloadTransferSize;
+		if (bandwidth <= stream->maxpsize) {
+			break;
+		}
+
+		probe->wKeyFrameRate = probe_min.wKeyFrameRate;
+		probe->wPFrameRate = probe_min.wPFrameRate;
+		probe->wCompQuality = probe_max.wCompQuality;
+		probe->wCompWindowSize = probe_min.wCompWindowSize;
+	}
+
+done:
+	return ret;
+}
+
+/****************************************************************************
+ * Name: uvc_video_decode
+ *
+ * Description:
+ *   Decode the data recevied on isoc/bulk endpoint based streaming device.
+ *
+ * Input Parameters:
+ *   a\buf - pointer to data buffer
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *   Probably called from an interrupt handler.
+ *
+ ****************************************************************************/
+static inline void uvc_video_decode(struct uvc_buf_object *bobj)
+{
+	uint8_t fid = 0;
+	uint8_t *pbuff;
+	uint8_t *trfbuff;
+	uint8_t hdrsize;
+	uint32_t pktsize;
+	FAR struct usbhost_uvc_state_s *dev = (FAR struct usbhost_uvc_state_s *)bobj->context;
+	struct uvc_streaming *stream = dev->stream;
+
+	trfbuff = (uint8_t *) bobj->buffer;
+	hdrsize = trfbuff[0];
+	/* A payload must be at least 2 bytes long, Header Length value must
+	 * be at least 2 bytes and can't be larger than the packet size.
+	 */
+	if (bobj->recsize < 2 || hdrsize < 2 || hdrsize > bobj->recsize) {
+		uvdbg("Recevied an invalid payload!!\n");
+		goto error;
+	}
+
+	fid = trfbuff[1] & UVC_STREAM_FID;
+
+	if (stream->last_fid != fid) {
+		stream->sequence++;
+		/* Discard the current frame and start fresh */
+		stream->trfsize = 0;
+		uvdbg("New Frame Detected:%d\n", stream->trfsize);
+	}
+
+	/* Mark the buffer as bad if the error bit is set. */
+	if (trfbuff[1] & UVC_STREAM_ERR) {
+		uvdbg(" Error in received payload.\n");
+		/* Discard the current frame and start fresh */
+		stream->trfsize = 0;
+		goto error;
+	}
+
+	pktsize = bobj->recsize - hdrsize;
+	if (pktsize > 0) {
+		/* Copy the data to user buffer */
+		pbuff = (uint8_t *) stream->reqbuff;
+		pbuff += stream->trfsize;
+		memcpy(pbuff, (trfbuff + hdrsize), pktsize);
+		/* Update received size */
+		stream->trfsize += pktsize;
+		if (stream->trfsize > stream->reqsize) {
+			DEBUGASSERT(0);
+		}
+	}
+
+	if (trfbuff[1] & UVC_STREAM_EOF) {
+		uvdbg("EOF Detected:%d\n", stream->trfsize);
+		stream->reqsize = 0;
+	}
+
+	stream->last_fid = fid;
+error:
+	return;
+}
+
+/****************************************************************************
+ * Name: uvc_transfer_thread
+ *
+ * Description:
+ *   Write worker thread which will process the recevied USB data, write the
+ *   data from USB transfer buffer to video frame.
+ *
+ * Input Parameters:
+ *   argc - argument countr
+ *   argv - array of arguments
+ *
+ * Returned Value:
+ *   Return always OK
+ *
+ ****************************************************************************/
+int uvc_transfer_thread(int argc, char *argv[])
+{
+	int ret = OK;
```

Comment:
I think we don't need to initialize `ret`.

Reply:
Done